engine-types/src/types.rs

```diff
@@ -314,10 +433,35 @@ impl<T> Stack<T> {
         self.stack
     }
 }
+
 pub fn str_from_slice(inp: &[u8]) -> &str {
     str::from_utf8(inp).unwrap()
 }
 
+pub mod error {
+    use crate::{fmt, String};
+
+    #[derive(Eq, Hash, Clone, Debug, PartialEq)]
+    pub enum BalanceOverflowError {
+        Overflow,
+    }
```

Comment:
```suggestion
    pub struct BalanceOverflowError;
```

An enum with a single variant seems unnecessarily complex to me compared with a simple unit struct.